h/api/test/storage_test.py

```diff
@@ -538,12 +538,83 @@ def annotation_data(self):
             }
         }
 
-    @pytest.fixture
-    def fetch_annotation(self, request):
-        patcher = patch('h.api.storage.fetch_annotation', autospec=True)
-        fetch_annotation = patcher.start()
-        request.addfinalizer(patcher.stop)
-        return fetch_annotation
+
+@pytest.mark.usefixtures('fetch_annotation')
+class TestDeleteAnnotationLegacy(object):
+
+    """Tests for delete_annotation() when the postgres feature flag is off."""
+
+    def test_it_only_fetches_the_legacy_annotation(self, fetch_annotation):
+        request = self.mock_request()
+
+        storage.delete_annotation(request, "test_id")
+
+        fetch_annotation.assert_called_once_with(
+            request, "test_id", _postgres=False)
+
+    def test_it_deletes_the_legacy_annotation(self, fetch_annotation):
+        storage.delete_annotation(self.mock_request(), "test_id")
+
+        fetch_annotation.return_value.delete.assert_called_once_with()
+
+    def mock_request(self):
+        request = DummyRequest()
+        request.feature = mock.Mock(return_value=False)
+        return request
+
+
+@pytest.mark.usefixtures('fetch_annotation')
+class TestDeleteAnnotation(object):
+
+    def test_it_fetches_the_annotation(self, fetch_annotation):
+        request = self.mock_request()
+
+        storage.delete_annotation(request, "test_id")
+
+        assert fetch_annotation.call_args_list[0] == mock.call(request,
+                                                               "test_id",
+                                                               _postgres=True)
+
+    def test_it_deletes_the_annotation(self, fetch_annotation):
+        request = self.mock_request()
+        first_return_value = mock.Mock()
+        second_return_value = mock.Mock()
+        fetch_annotation.side_effect = [
+            first_return_value,
+            second_return_value,
+        ]
```

Comment:
We need to make sure it deletes the Postgres annotation not the Es one so it's necessary to distinguish between `fetch_annotation()`'s first and second return values.